### PR TITLE
Navbar: Add Badge icon to Notification Button 

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
## Description

- Added prop counter as a badge to the notification icon button

References #
Fixes: #1011 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx [hunk 1]
```diff
@@ -70,6 +70,9 @@ export class ModusNavbar {
   /** (optional) Whether to show notifications. */
   @Prop() showNotifications: boolean;
 
+  /** (optional) Whether to show badge on top of notification */
+  @Prop() counter: string;
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/modus-navbar.tsx:74]: Can we make this `notificationCount` as a number?

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx [hunk 2]
```diff
@@ -361,6 +364,24 @@ export class ModusNavbar {
...
+        } else if (counter > 99) {
+          counterValue = '99+';
+        } else {
+          counterValue = this.counter;
+        }
+      } else {
+        this.counter = null;
+      }
+    }
+    console.log('countervalue', counterValue);
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/modus-navbar.tsx:384]: Remove `console` logs

Also this above logic can probably be simplified if the `prop` is a `number`

Also break this into a separate function, render is getting a little heavy with all this logic.  something like 
```js
const notificationCount = getNotificationCount()`
```
That can return a string value

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx [hunk 3]
```diff
@@ -109,14 +115,16 @@ const defaultApps = [{
   url: 'https://modus.trimble.com/',
 }];
 
-const Template = ({ buttons, enableSearchOverlay, helpTooltip, profileMenuOptions, searchTooltip, showHelp, showProfile, showSearch }) => html`
+const Template = ({ buttons,counter, enableSearchOverlay, helpTooltip, profileMenuOptions, searchTooltip, showHelp, showProfile, showSearch }) => html`
```
Comment (cjwinsor) [inline stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx:118]: missing a space after the `,` and before `counter`

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx [hunk 4]
```diff
@@ -174,16 +182,18 @@ Default.args = {
   showHelp: false,
   showProfile: true,
   showSearch: false,
+  counter:""
 };
 
-const FailedToLoadAvatarTemplate = ({ buttons, enableSearchOverlay, helpTooltip, profileMenuOptions, searchTooltip, showHelp, showProfile, showSearch }) => html`
+const FailedToLoadAvatarTemplate = ({ buttons,counter, enableSearchOverlay, helpTooltip, profileMenuOptions, searchTooltip, showHelp, showProfile, showSearch }) => html`
```
Comment (cjwinsor) [inline stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx:188]: missing a space after the , and before counter

**Review by cjwinsor — CHANGES_REQUESTED** [review]

**Comment by cjwinsor** [comment]
@coliff @egunther39 Can you take a look and let me know if the size of this notification badge is too small in the deploy preveiw?...

File: stencil-workspace/src/components/modus-navbar/modus-navbar.vars.scss [hunk 5]
```diff
@@ -58,3 +58,6 @@ $modus-navbar-blue-profile-icon-active-border-color: var(
   --modus-navbar-blue-profile-icon-active-border-color,
   #019aeb
 ) !default;
+
+// Badge
+$modus-navbar-badge-bg: var(--modus-navbar-badge-bg, #da212c) !default;
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/modus-navbar.vars.scss:63]: Badge is showing as blue in deploy preview

**Comment by egunther39** [comment]
@cjwinsor - It's a bit small. Here is an example of the red badge we use in our application. The badge is 20x20 inside/overlaid above the 48x48 navbar button/icon. I'd suggest, doing the same.
image

**Comment by cjwinsor** [comment]
Here is a shot with a medium badge
image

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-navbar/modus-navbar.tsx:74]: Isn't string preferable so in the future we can allow up to 3 character string

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx]: Formatted

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx]: Formatted

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-navbar/modus-navbar.tsx]: Modified

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-navbar/modus-navbar.tsx]: Made it to be a number for now

**Review by cjwinsor — CHANGES_REQUESTED** [review]

**Comment by cjwinsor** [comment]
@ElishaSamPeterPrabhu Notifications are still blue in the deploy preview as well.

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar-storybook-docs.mdx [hunk 6]
```diff
@@ -223,6 +223,7 @@ interface ModusNavbarButton {
 | ---------------------- | ------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------- | ----------- |
 | `apps`                 | --                       | (optional) The apps to render in the apps menu.                                                                                                                                         | `ModusNavbarApp[]`        | `undefined` |
 | `buttons`              | --                       | (optional) The buttons to render in the Navbar.                                                                                                                                         | `ModusNavbarButton[]`     | `undefined` |
+| `notificationCount`    | `notification-count`     | (optional) To add the counter value to the notification icon.                                                                                                                           | `string`                  | `undefined` |
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar-storybook-docs.mdx:226]: Updated

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx [hunk 7]
```diff
@@ -64,7 +64,15 @@ export default {
...
         type: { summary: 'ModusNavbarTooltip' },
       },
-    }
+    },
+    notificationCount:{
+      control: { type: 'text' },
+      name: 'notification-count',
+      description: 'To add the counter value to the notification icon',
+      table: {
+        type: { summary: 'string' },
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx:73]: Updated

**Comment by ElishaSamPeterPrabhu** [comment]
> @ElishaSamPeterPrabhu Notifications are still blue in the deploy preview as well.

Now its red

> @ElishaSamPeterPrabhu Notifications are still blue in the deploy preview as well.

Its red now

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**